crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -231,11 +304,16 @@ impl C8YHttpProxyActor {
         software_list: C8yUpdateSoftwareListResponse,
     ) -> Result<Unit, C8YRestError> {
         let url = self.end_point.get_url_for_sw_list();
-        let req_builder = HttpRequestBuilder::put(url)
-            .header("Accept", "application/json")
-            .header("Content-Type", "application/json")
-            .json(&software_list);
-        let http_result = self.execute(req_builder).await?;
+        let req_builder_closure =
+            |token: String, internal_id: String| -> Result<HttpRequestBuilder, C8YRestError> {
+                let url = update_url_with_new_internal_id(internal_id, &url)?;
```

Comment:
Instead of having this update function, I have a feeling that the `self.end_point.get_url_for_sw_list()` and other `get_url` functions in there should have taken the internal id as an argument, so that this invocation was cleaner.

Reply:
Here we don't need to even change the signature of the API, it can work as it is because it picks up the internal ID from the `endpoint`. But the issue here is I can't use the `self.end_point.get_url_for_sw_list()` inside the closure as it takes the immutable self. And later the `execute_retry` takes the `mutable self` which leads to an error.